### DISCUSSION TITLE
How to find number of selected files

### DISCUSSION THREAD

**Discussion by linidu** [rpldy/react-uploady#226] [Q&A]
I'm using this uploady for select multiple images. Here I'm using uploady like this.

            <Uploady
                destination={{ url: "my-server.com/upload" }}
                fileFilter={filterBySize}
                accept="image/*"
                multiple
            >
                <div className="App">
                    <UploadButton />
                </div>
            </Uploady>

And inside filerBySize function..
const filterBySize = (file) => {
        images.push(file);
};

I want to set this images array to state. For set that into the state, I want to know how many images are there in the selected list.

Is there any way to find that selected count using uploady.

**Comment by yoavniran** [comment]
Hey @linidu 
fileFilter isnt a good place to manage your state since its called per item whenever a batch is added.
Keep in mind that the [@rpldy/uploader](https://github.com/rpldy/react-uploady/tree/master/packages/core/uploader) (used by Uploady) can keep track of multiple batches and all of their items. It doesnt provide you with the info about the item being finished either.

The better strategy would be to register to the item events (or use the corresponding hooks): [useItemStartListener]( https://github.com/rpldy/react-uploady/tree/master/packages/ui/uploady#useitemstartlistener-event-hook) & [useItemFinalizeListener](https://github.com/rpldy/react-uploady/tree/master/packages/ui/uploady#useitemfinalizelistener-event-hook)


The first is called when an item is added to the upload queue. The latter is called whenever an item is done. Either because it uploaded successfully or in case of failure.

This way you can update your state whenever item begins and remove it when its done.

**Reply by linidu** [reply]
Hi @yoavniran ,
Thank you very much for the explanation. I'll try in that way.
